insert.go

```diff
@@ -256,3 +295,71 @@ func (b InsertBuilder) SetMap(clauses map[string]interface{}) InsertBuilder {
 func (b InsertBuilder) Select(sb SelectBuilder) InsertBuilder {
 	return builder.Set(b, "Select", &sb).(InsertBuilder)
 }
+
+// OnConflictKey specify the primary key or columns for the ON CONFLICT DO UPDATE SET clause.
+func (b InsertBuilder) OnConflictKey(columns ...string) InsertBuilder {
+	for _, col := range columns {
+		b = builder.Append(b, "OnConflictKey", col).(InsertBuilder)
+	}
+	return b
+}
+
+// OnConflictSet adds ON CONFLICT DO UPDATE SET clause to the query.
+func (b InsertBuilder) OnConflictSet(column string, value interface{}) InsertBuilder {
+	return builder.Append(b, "OnConflictSetClauses", setClause{column: column, value: value}).(InsertBuilder)
+}
+
+// OnConflictSetSetMap is a convenience method which calls .Set for each key/value pair in clauses.
+func (b InsertBuilder) OnConflictSetSetMap(clauses map[string]interface{}) InsertBuilder {
+	keys := make([]string, len(clauses))
+	i := 0
+	for key := range clauses {
+		keys[i] = key
+		i++
+	}
+	sort.Strings(keys)
+	for _, key := range keys {
+		val, _ := clauses[key]
+		b = b.OnConflictSet(key, val)
+	}
+	return b
+}
+
+// OnConflictFormat
+
+// OnConflictFormat interface. Encapsulate different phrases used for ON CONFLICT.
+type OnConflictFormat interface {
+	String(cols []string) string
+}
+
+// Global variables for different OnConflictFormats
+var (
+	OnConflict = onConflictFormat{}
+
+	OnConflictWithKey = onConflictWithKeyFormat{}
+
+	OnDuplicate = onDuplicateFormat{}
+)
+
+type onConflictFormat struct{}
+type onConflictWithKeyFormat struct{}
+type onDuplicateFormat struct{}
+
+// String generates the string: ON CONFLICT DO UPDATE SET
+func (onConflictFormat) String(cols []string) string {
+	return " ON CONFLICT DO UPDATE SET "
+}
+
+// String generates the string: ON CONFLICT(col1,col2) DO UPDATE SET
+func (onConflictWithKeyFormat) String(cols []string) string {
+	key := ""
+	if len(cols) > 0 {
+		key = fmt.Sprintf(" (%s)", strings.Join(cols, ","))
```

Comment:
Don't column names need escaping?

I mean, they do, but does squirrel give you the tools to do it?

If so, you should use them here, and in any other place where you use format strings

Reply:
mmm... I think Squirrel does not have columna name escaping support. I don't think it'd be convenient to add it everywhere. 

I'm tempted to do my own multi-dialect query builder, but it's not the time